analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundantNullCheck.CSharp9.cs

```diff
@@ -1,29 +1,227 @@
-﻿object n = 5;
+﻿object a = "";
+object b = "";
+object n = 5;
+object m = 5;
+Apple apple = new();
 
+// AND
 if (n != null && n is nint) // Noncompliant
 {
 }
 
-if (n is not null && n is nint) // Compliant - FN
+if (n is not null && n is nint nintValue) // Noncompliant
 {
 }
 
-if (n is not null && n is not null) // Compliant - FN
```

Comment:
for the record: I removed this test case because this rule should not raise an issue in this case (consistency with C# 7 test cases)